kube-core/src/resource.rs

```diff
@@ -184,7 +184,9 @@ pub trait ResourceExt: Resource {
     /// For more precise use cases, prefer doing your own defaulting.
     /// For true uniqueness, prefer [`uid`](ResourceExt::uid).
     fn name_any(&self) -> String;
-
+    /// Returns name of the CustomResourceDefinition object exposing this resource.
+    /// If this resource has other nature (builtin or ApiService), returns unspecified value.
+    fn crd_name(dt: &Self::DynamicType) -> String;
```

Comment:
Well, in fact I only added this function because it feels useful in the example.

> Is it not possible to rely on CustomResourceExt instead?
In that case one can not perform migration for non-CustomResource CRD (or CRD with name known only in runtime). However, such a case sounds very special, probably OK to leave it unsupported.

Reply:
I'm marking this as resolved because now we have an API which is both non-confusing and supports all kinds of CRDs.